mlflow/models/evaluation/base.py

```diff
@@ -1247,6 +1247,44 @@ def predict(self, context, model_input: pd.DataFrame):
     return _PythonModelPyfuncWrapper(python_model, None, None)
 
 
+def _get_model_from_deployment_endpoint_uri(
+    endpoint_uri: str, params: Optional[Dict[str, Any]] = None
+):
+    from mlflow.metrics.genai.model_utils import _call_deployments_api, _parse_model_uri
+    from mlflow.pyfunc.model import _PythonModelPyfuncWrapper
+
+    class ModelFromDeploymentEndpoint(mlflow.pyfunc.PythonModel):
+        def __init__(self, endpoint, params):
+            self.endpoint = endpoint
+            self.params = params
+
+        def predict(self, context, model_input: pd.DataFrame):
+            if "inputs" not in model_input.columns:
```

Comment:
What happens if the user passes `data` to `mlflow.evaluate()` of type list and then uses (or does not use) the `feature_names` argument? Does this fail? We should definitely support passing a list & test that case as well

Reply:
I agree that we should support passing list input, but the usage of `feature_names` is still not clear to me after reading the docs tbh.... let me clarify my understanding from reading codes:
* When `feature_names` is provided for DataFrame input, those columns are picked to prediction input.
* When `feature_names` is NOT provided for DataFrame input, all columns except `target` will be picked to input.
* When `feature_names` is provided for list input, the list will be converted to dataframe with those given column names.
* When `feature_names` is NOT provided for list input, the list will be input with automatically generated column names `feature_names_0`, ...

Is my understanding correct? If my understanding is correct, supporting them is not super hard, with some limitation (like `feature_names` should only contain single input column).

Reply:
Added support for list input in the latest revision. One friction I found is that it doesn't accepts the flat list like `["What is MLflow?", "What is Spark"]`, instead each element has to be wrapped in another list like `[["What is MLflow?"], ["What is Spark?"]]`. This is not specific to endpoint evaluation.